incubator/sparkoperator/templates/webhook-init-job.yaml

```diff
@@ -2,8 +2,11 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: {{ include "sparkoperator.fullname" . }}-init
+  name: {{ include "sparkoperator.fullname" . }}-webhook-init
   namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
```

Comment:
Why post-install? And you've tested that post-upgrade will make it no longer necessary to manually delete the job before upgrade?

Reply:
It must be post-install to ensure the credentials are in place to create the secret.

Upgrading to this version will still require the job to be deleted the first time. Subsequent deploys will not leave the init job around. I had contemplated deleting the init job as part of the cleanup job hook, but making the init a hook seems like a more elegant solution.

Reply:
What do you mean by "ensure the credentials are in place to create the secret"?

Reply:
The init hook creates the `spark-webhook-certs` secret. To do that it must have credentials in the cluster and permissions to create secrets. It uses the credentials/permissions of the `sparkoperator` service account. The `sparkoperator` service account isn't created until the chart is "installed".

Here is the script that runs in the init job:
https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/hack/gencerts.sh

Reply:
The init job will wait for the secret to be created. So this doesn't actually pose an issue in practice.

Reply:
I had to remove these annotations to fix the problem that it caused in our deployments via argo-cd.

the problem was spark-operator containers were dying due to lack of spark-webhook-secrets, and spark-webhook-secrets was never be able to be created because deployment was never completed and helm post-install hooks were never executed.

Reply:
I also had to remove these annotations for exactly the reasons as @ekesken .

As far as I understand, because of this change, it is now impossible to install the spark-operator with webhook support using this chart.